File: Donggyo/week4/day17/가장큰정사각형찾기.java
class Solution
{
    public int solution(int [][]board)
    {
        int answer = -1;
        int max_length = 0;
        int R = board.length;
        int C = board[0].length;
        int[][] dp = new int[R][C];

        if(R<=1 && C<=1) return board[0][0]*board[0][0];
        
        for(int c=0;c<C;c++){
            dp[0][c] = board[0][c];
        }
        for(int r=0; r<R;r++){
            dp[r][0] = board[r][0];
        }
            
        for(int r = 1; r<R; r++){
            for(int c = 1; c<C; c++){
                if(board[r][c]==1){
                    dp[r][c] = Math.min(Math.min(dp[r-1][c-1], dp[r-1][c]),dp[r][c-1])+1;
                    max_length = Math.max(dp[r][c], max_length);
                }
            }
        }
        answer = max_length*max_length;
        return answer;
    }
}

File: Jiho/week4/day17/가장 큰 정사각형 찾기.js
function solution(board) {
  let answer = 0;
  const dp = board.map((v) => v);

  if (board.length === 1 && board[0].length === 1 && board[0][0] === 1)
    return 1;

  for (let i = 1; i < board.length; i++) {
    for (let j = 1; j < board[0].length; j++) {
      if (board[i][j] !== 0) {
        dp[i][j] = Math.min(dp[i - 1][j], dp[i - 1][j - 1], dp[i][j - 1]) + 1;
        if (dp[i][j] > answer) answer = dp[i][j];
      }
    }
  }

  return answer * answer;
}
